microbench/src/main/java/io/netty/microbench/buffer/ByteBufAllocatorAllocPatternBenchmark.java

```diff
@@ -0,0 +1,568 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.microbench.buffer;
+
+import io.netty.buffer.AdaptiveByteBufAllocator;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.MiByteBufAllocator;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.internal.MathUtil;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.SplittableRandom;
+
+@State(Scope.Thread)
+@Warmup(iterations = 10, time = 1)
+@Measurement(iterations = 10, time = 1)
+@Threads(8)
+public class ByteBufAllocatorAllocPatternBenchmark extends AbstractMicrobenchmark {
+
+    private static final PooledByteBufAllocator pooledAlloc = PooledByteBufAllocator.DEFAULT;
+    private static final ByteBufAllocator adaptiveAllocator = new AdaptiveByteBufAllocator();
+    private static final MiByteBufAllocator miMallocAllocator = new MiByteBufAllocator();
+
+    private static final int SEED = 42;
+    // Allocation size list.
+    private static final ArrayList<Integer> sizeList = new ArrayList<>();
```

Comment:
This one create lot of noise in the benchmarks...

Reply:
@laosijikaichele instead of a double data-dependent look up here 

```java
    private void directAlloc(Blackhole blackhole, ByteBufAllocator alloc, ByteBuf[] buffers) {
        int sizeIndex = sizeIndexes[getNextSizeIndex()];
        int size = sizeList.get(sizeIndex);
        // 
```
why not storing directly the sizes into a short[] and use such?

Reply:
Changed ArrayList to primitive array, and use one look up, instead of double look up for sizes. on this https://github.com/netty/netty/pull/15525/commits/4b8188828ffaf9d16a49a1ad055c01643a689749